pkg/controllers/streaming/processor_controller.go

```diff
@@ -455,16 +455,20 @@ func (r *ProcessorReconciler) resolveStreams(ctx context.Context, processorCoord
 			Name:      binding.Stream,
 		}
 		var stream streamingv1alpha1.Stream
-		// track stream for new coordinates
-		r.Tracker.Track(
-			tracker.NewKey(stream.GetGroupVersionKind(), streamNSName),
-			processorCoordinates,
-		)
 		if err := r.Client.Get(ctx, streamNSName, &stream); err != nil {
 			return nil, nil, err
+		} else if !stream.Status.IsReady() {
+			return nil, nil, fmt.Errorf("stream %q is not Ready", streamNSName)
```

Comment:
What should happen if the processor was previously created when a stream becomes unready?

Reply:
Good question, although I don't think we can witness that in practice now, unless we're really trying to.
I would argue that the deployments should be removed, wdyt?

Reply:
I'm fine with removing/not creating the processor when any stream is not ready. We may need to refine that behavior down the road, but we can cross the bridge when we get to it.

Reply:
Please TAL @ 97bf58c